tests/Avalonia.Visuals.UnitTests/Media/TextFormatting/TestDataGenerator.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+
+namespace Avalonia.Visuals.UnitTests.Media.TextFormatting
+{
+    public abstract class TestDataGenerator : IEnumerable<object[]>
+    {
+        private readonly string _fileName;
+        private readonly List<object[]> _testData;
+
+        protected TestDataGenerator(string fileName)
+        {
+            _fileName = fileName;
+            _testData = ReadTestData();
+        }
+
+        public IEnumerator<object[]> GetEnumerator()
+        {
+            return _testData.GetEnumerator();
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        private List<object[]> ReadTestData()
+        {
+            var testData = new List<object[]>();
+
+            using (var client = new HttpClient())
+            {
+                var url = Path.Combine(UnicodeDataGenerator.Ucd, _fileName);
+
+                using (var result = client.GetAsync(url).GetAwaiter().GetResult())
+                {
+                    if (!result.IsSuccessStatusCode)
+                        return testData;
+
+                    using (var stream = result.Content.ReadAsStreamAsync().GetAwaiter().GetResult())
+                    using (var reader = new StreamReader(stream))
+                    {
+                        while (!reader.EndOfStream)
+                        {
+                            var line = reader.ReadLine();
+
+                            if (line == null)
+                            {
+                                break;
+                            }
+
+                            if (line.StartsWith("#") || string.IsNullOrEmpty(line))
+                            {
+                                continue;
+                            }
+
+                            var elements = line.Split('#');
+
+                            elements = elements[0].Replace("÷\t", "÷").Trim('÷').Split('÷');
```

Comment:
this could be a bit of a pain in the ass to do but can you encode all the unicode char literals into \uXXXX format? just to avoid potential encoding issues

Reply:
if it's too much work then ignore this suggestion :)

Reply:
This is just a generator. Didn't bother optimizing things because it is only run when the Unicode spec changes. I will try this on Linux soon but a change to this should not delay the PR.

Reply:
alright, i agree